app/services/SessionCacheService.scala

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import common.Dates.constructDate
+import common.KeystoreKeys.ResidentPropertyKeys
+import connectors.SessionCacheConnector
+import models.resident._
+import models.resident.properties._
+import models.resident.properties.gain.{OwnerBeforeLegislationStartModel, WhoDidYouGiveItToModel, WorthWhenGiftedModel}
+import play.api.mvc.Results.Redirect
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.frontend.exceptions.ApplicationException
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+object SessionCacheService extends SessionCacheService{
+   lazy val sessionCacheConnector = SessionCacheConnector
+}
+
+trait SessionCacheService {
+  val sessionCacheConnector: SessionCacheConnector
+  lazy val homeLink = sessionCacheConnector.homeLink
+
+  //scalastyle:off
+  def getPropertyGainAnswers(implicit hc: HeaderCarrier): Future[YourAnswersSummaryModel] = {
+    val disposalDate = sessionCacheConnector.fetchAndGetFormData[DisposalDateModel](ResidentPropertyKeys.disposalDate).map(formData =>
+      constructDate(formData.get.day, formData.get.month, formData.get.year))
+
+    //This is a proposed alternate method of writing the map without needing the case statement, need a judgement on whether
+    //to use this method or older ones. Fold automatically handles the None/Some cases without matching manually
+    val disposalValue = sessionCacheConnector.fetchAndGetFormData[DisposalValueModel](ResidentPropertyKeys.disposalValue)
+      .map(_.fold[Option[BigDecimal]](None)(input => Some(input.amount)))
```

Comment:
Could this not be written as .map(_.map(_.amount))?

Reply:
Or flatMap, maybe?